script.js

```diff
@@ -1,12 +1,124 @@
-//You can edit ALL of the code here
+// INITIALISATION OF COMMON ELEMENTS
+const rootElem = document.getElementById("root"),
+  searchBar = document.getElementById("searchBar"),
+  matches = document.getElementById("matches"),
+  select = document.getElementById("selectEpisode");
+let EPISODES;
+
+// PROXY FUNCTION FOR RETURNING CACHED DATA FROM PROMISE
+const getAllEpisodes = () => {
+  return EPISODES;
+};
+
+// INITIALISATION CALLED BY WINDOW.ONLOAD WITH ADDED CATCHED DATA PROVIDED BY THE FETCH API
 function setup() {
-  const allEpisodes = getAllEpisodes();
-  makePageForEpisodes(allEpisodes);
+  fetch("https://api.tvmaze.com/shows/82/episodes")
+    .then((response) => response.json())
+    .then((data) => {
+      EPISODES = data;
+      render(getAllEpisodes());
+    })
+    .catch((error) => {
+      console.log("Data hasn't arrive see error: ", error);
+    });
 }
 
 function makePageForEpisodes(episodeList) {
-  const rootElem = document.getElementById("root");
   rootElem.textContent = `Got ${episodeList.length} episode(s)`;
 }
 
-window.onload = setup;
+const episodeContainsTerm = (episode, searchTerm) => {
+  const inTitle = episode.name,
+    inDescription = episode.summary;
+  return searchTerm.test(inTitle) || searchTerm.test(inDescription);
+};
+
+const searchCollection = () => {
+  let searchTerm;
+  if (!searchBar.value) {
+    rootElem.replaceChildren();
+    render(getAllEpisodes());
+    return;
+  } else {
+    searchTerm = new RegExp(searchBar.value, "i");
+    let episodesFound = getAllEpisodes().filter((episode) => {
+      return episodeContainsTerm(episode, searchTerm);
+    });
+    rootElem.replaceChildren();
+    render(episodesFound);
+    return;
+  }
+};
+
+const minTwoDigits = (num) => {
```

Comment:
this is great, but JavaScript has a [padStart method](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/padStart) which will make your life much easier in case you need to deal with more 0s